anta/tests/routing/bgp.py

```diff
@@ -447,3 +448,204 @@ def test(self) -> None:
 
         if failures:
             self.result.is_failure(f"Failures: {list(failures.values())}")
+
+
+class VerifyBGPPeerMPCaps(AntaTest):
+    """
+    Verifies the multiprotocol capabilities of a BGP peer in a specified VRF.
+    Expected results:
+        * success: The test will pass if the BGP peer's multiprotocol capabilities are advertised, received, and enabled in the specified VRF.
+        * failure: The test will fail if BGP peers are not found or multiprotocol capabilities are not advertised, received, and enabled in the specified VRF.
+    """
+
+    name = "VerifyBGPPeerMPCaps"
+    description = "Verifies the multiprotocol capabilities of a BGP peer in a specified VRF"
+    categories = ["routing", "bgp"]
+    commands = [AntaCommand(command="show bgp neighbors vrf all")]
+
+    class Input(AntaTest.Input):
+        """
+        Input parameters of the testcase.
+        """
+
+        bgp_peers: List[BgpPeers]
+        """List of BGP peers"""
+
+        class BgpPeers(BaseModel):
+            """
+            This class defines the details of a BGP peer.
+            """
+
+            peer_address: IPv4Address
+            """IPv4 address of a BGP peer"""
+            vrf: str = "default"
+            """Optional VRF for BGP peer. If not provided, it defaults to `default`."""
+            capabilities: List[MultiProtocolCaps]
+            """Multiprotocol capabilities"""
+
+    @AntaTest.anta_test
+    def test(self) -> None:
+        failures: dict[str, Any] = {}
+
+        # Iterate over each bgp peer
+        for bgp_peer in self.inputs.bgp_peers:
+            peer = str(bgp_peer.peer_address)
+            vrf = bgp_peer.vrf
+            capabilities = bgp_peer.capabilities
+
+            # Check if BGP output exists
+            if (
+                not (bgp_output := get_value(self.instance_commands[0].json_output, f"vrfs.{vrf}.peerList"))
+                or (bgp_output := get_item(bgp_output, "peerAddress", peer)) is None
```

Comment:
I think I saw issues in the past with walrus operator `:=` and `or`. Did you test properly? I might be wrong.

Reply:
write now its giving me proper result. I tested again with different inputs